src/Client/TestHint.cpp

```diff
@@ -6,25 +6,46 @@
 #include <IO/ReadHelpers.h>
 #include <Parsers/Lexer.h>
 
+namespace DB::ErrorCodes
+{
+extern const int CANNOT_PARSE_TEXT;
+}
+
 namespace
 {
 
 /// Parse error as number or as a string (name of the error code const)
-int parseErrorCode(DB::ReadBufferFromString & in)
+DB::TestHint::error_vector parseErrorCode(DB::ReadBufferFromString & in)
 {
-    int code = -1;
-    String code_name;
+    DB::TestHint::error_vector error_codes{};
 
-    auto * pos = in.position();
-    tryReadText(code, in);
-    if (pos != in.position())
+    while (!in.eof())
     {
-        return code;
+        int code = -1;
+        String code_name;
+        auto * pos = in.position();
+
+        tryReadText(code, in);
+        if (pos == in.position())
+        {
+            readStringUntilWhitespace(code_name, in);
+            code = DB::ErrorCodes::getErrorCodeByName(code_name);
+        }
+        error_codes.push_back(code);
+
+        if (in.eof())
+            break;
+        skipWhitespaceIfAny(in);
+        if (in.eof())
+            break;
+        char c;
+        in.readStrict(c);
+        if (c != '|')
```

Comment:
We discussed and we would prefer to have `,` instead of `|` as separator.

Reply:
I didn't put any thought to it. I used `|` because it felt natural (or) but I don't have a strong opinion about it. It's just for testing purposes so any sensible character seems good

Reply:
I'll change it on Monday or feel free to change it in the PR before that

Reply:
I've added a new commit reworking how the comment is parsed. It now uses a lexer which is simpler to reason about and supports using `,` both next to the error and with or without spaces.